pubsub/subscriber.go

```diff
@@ -0,0 +1,96 @@
+package pubsub
+
+import (
+	"cloud.google.com/go/pubsub"
+	"context"
+	"encoding/json"
+	"errors"
+	queuesgo "github.com/merlinapp/queues-go"
+	"log"
+	"strconv"
+)
+
+type subscriber struct {
+	project          string
+	subscriptionName string
+	elements         []routerElement
+}
+
+type routerElement struct {
+	event       string
+	handlerFunc queuesgo.HandlerFunc
+}
+
+/*
+Creates a new Google's pubsub subscriber implementation
+the subscription name must exists already on the given projects
+*/
+func NewSubscriber(project, subscriptionName string) queuesgo.Subscriber {
+	return &subscriber{
+		project:          project,
+		subscriptionName: subscriptionName,
+	}
+}
+
+func (p *subscriber) RegisterFunction(eventName string, handler queuesgo.HandlerFunc) error {
+	if eventName == "" {
+		return errors.New("invalid event name")
+	}
+	p.elements = append(p.elements, routerElement{event: eventName, handlerFunc: handler})
+	return nil
+}
+
+func (p *subscriber) Subscribe(ctx context.Context) error {
+	pubsubClient, _ := pubsub.NewClient(ctx, p.project)
+	sub := pubsubClient.Subscription(p.subscriptionName)
+	err := sub.Receive(ctx, func(ctx context.Context, message *pubsub.Message) {
+		log.Printf("Recieved message: %s", message.Data)
```

Comment:
minor - typo - `Recieve` is a funny word - it's actually spelt `Received` even though that's not how it sounds :)

Reply:
Done, thanks!